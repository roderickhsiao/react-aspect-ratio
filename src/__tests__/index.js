import React from 'react';
import renderer from 'react-test-renderer';

import OldAspectRatio, { AspectRatio } from '../index'; // // eslint-disable-line import/no-named-as-default

describe('Aspect Ratio', () => {
  describe('React < 15.6', () => {
    it('should render wrapper for children', () => {
      const innerImage = <img src="https://foo.bar" alt="demo" />;
      const node = renderer
        .create(<OldAspectRatio ratio="4/3">{innerImage}</OldAspectRatio>)
        .toJSON();
      expect(node).toMatchSnapshot();
    });

    it('should support number as props', () => {
      const innerImage = <img src="https://foo.bar" alt="demo" />;
      const node = renderer
        .create(<OldAspectRatio ratio={0.75}>{innerImage}</OldAspectRatio>)
        .toJSON();
      expect(node).toMatchSnapshot();
    });

    it('custom style willl be adpoted', () => {
      const innerImage = <img src="https://foo.bar" alt="demo" />;
      const node = renderer
        .create(
          <OldAspectRatio ratio="4/3" style={{ color: '#fff' }}>
            {innerImage}
          </OldAspectRatio>
        )
        .toJSON();

      expect(node).toMatchSnapshot();
    });
  });

  describe('React > 15.6', () => {
    it('should render wrapper for children', () => {
      const innerImage = <img src="https://foo.bar" alt="demo" />;
      const node = renderer.create(<AspectRatio ratio="4/3">{innerImage}</AspectRatio>).toJSON();
      expect(node).toMatchSnapshot();
    });

    it('should support number as props', () => {
      const innerImage = <img src="https://foo.bar" alt="demo" />;
      const node = renderer.create(<AspectRatio ratio={0.75}>{innerImage}</AspectRatio>).toJSON();
      expect(node).toMatchSnapshot();
    });

    it('custom style willl be adpoted', () => {
      const innerImage = <img src="https://foo.bar" alt="demo" />;
      const node = renderer
        .create(
          <AspectRatio ratio="4/3" style={{ color: '#fff' }}>
            {innerImage}
          </AspectRatio>
        )
        .toJSON();

      expect(node).toMatchSnapshot();
    });
  });
});
