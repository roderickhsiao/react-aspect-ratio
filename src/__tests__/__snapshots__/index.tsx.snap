// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Aspect Ratio React < 15.6 custom style will be adopted 1`] = `
<div
  className="react-aspect-ratio-placeholder"
  style={
    {
      "--aspect-ratio": "(4/3)",
      "color": "#fff",
    }
  }
>
  <img
    alt="demo"
    src="https://foo.bar"
  />
</div>
`;

exports[`Aspect Ratio React < 15.6 should render wrapper for children 1`] = `
<div
  className="react-aspect-ratio-placeholder"
  style={
    {
      "--aspect-ratio": "(4/3)",
    }
  }
>
  <img
    alt="demo"
    src="https://foo.bar"
  />
</div>
`;

exports[`Aspect Ratio React < 15.6 should support number as props 1`] = `
<div
  className="react-aspect-ratio-placeholder"
  style={
    {
      "--aspect-ratio": "(0.75)",
    }
  }
>
  <img
    alt="demo"
    src="https://foo.bar"
  />
</div>
`;

exports[`Aspect Ratio React > 15.6 custom style will be adopted 1`] = `
<div
  className="react-aspect-ratio-placeholder"
  style={
    {
      "--aspect-ratio": "(4/3)",
      "color": "#fff",
    }
  }
>
  <img
    alt="demo"
    src="https://foo.bar"
  />
</div>
`;

exports[`Aspect Ratio React > 15.6 should render wrapper for children 1`] = `
<div
  className="react-aspect-ratio-placeholder"
  style={
    {
      "--aspect-ratio": "(4/3)",
    }
  }
>
  <img
    alt="demo"
    src="https://foo.bar"
  />
</div>
`;

exports[`Aspect Ratio React > 15.6 should support number as props 1`] = `
<div
  className="react-aspect-ratio-placeholder"
  style={
    {
      "--aspect-ratio": "(0.75)",
    }
  }
>
  <img
    alt="demo"
    src="https://foo.bar"
  />
</div>
`;
